storage/fsStorage.go

```diff
@@ -1,20 +1,84 @@
 package storage
 
-import "fmt"
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+)
 
-type FsStorage struct {
+type fsStorage struct {
 	path string
 }
+type myStruct struct {
+	Key   string
+	Value string
+}
+
+func (st fsStorage) Read(key string) (string, error) {
+	allDataFromJson := []myStruct{}
+	file, err := ioutil.ReadFile(st.path)
+	if err != nil {
+		return "", err
+	}
+	json.Unmarshal(file, &allDataFromJson)
+	if err != nil {
+		return "", err
+	}
+	for _, v := range allDataFromJson {
+		if v.Key == key {
+			return v.Value, nil
+		}
+	}
+	return "", errors.New("item was not found")
+}
+func (st fsStorage) Write(key string, value string) error {
+	err := checkFile(st.path)
+	if err != nil {
+		return err
+	}
+
+	file, err := ioutil.ReadFile(st.path)
+	if err != nil {
+		return err
+	}
 
-func (st FsStorage) Read(key string) (string, error) {
-	return "got it", nil
+	data := []myStruct{}
+
+	json.Unmarshal(file, &data)
```

Comment:
The same about error.

Reply:
How is that resolved? The same error is checked two times, what's the point?

Reply:
?

Reply:
resolved in the l[ast commit](https://github.com/retttddd/golangProjectv3/pull/1/commits/44d0c73d36187fa2398318dd7e5b9cc1184551a9)